addons/mrp/models/mrp_production.py

```diff
@@ -1970,7 +1970,7 @@ def _default_amounts(production):
                 workorder.duration_expected = workorder._get_duration_expected()
 
             # Adapt quantities produced
-            for workorder in production.workorder_ids:
+            for workorder in production.workorder_ids.sorted('id'):
```

Comment:
Not sure about this. If I manually change the start date of workorder on the original MO, the order will change, Sorting by `id` here will make the old workorders mismatch the new ones